oca_port/squash_bot_commits.py

```diff
@@ -0,0 +1,186 @@
+        choice = click.prompt("Select a commit to squash into:", type=int)
+        if not choice:
+            self.skipped_commits.append(commit)
+            return False
+        selected_commit = available_commits[choice - 1]
+        reorder = selected_commit.hexsha != commit.parents[0]
+        return self._squash(commit, selected_commit, reorder)
+
+    def _squash(self, commit, target_commit, reorder=False):
+        base_commit = target_commit.parents[0]
+        confirm = "\n".join(
+            [
+                "\nCommits to Squash:",
+                f"\t{bc.OKCYAN}{commit.hexsha[:9]}{bc.ENDC} {commit.summary}",
+                f"\t{bc.OKCYAN}{target_commit.hexsha[:9]}{bc.ENDC} {target_commit.summary}\n",
+            ]
+        )
+        if not click.confirm(confirm):
+            return False
+        editor_script = ""
+        if reorder:
+            with tempfile.NamedTemporaryFile(delete=False, mode="w") as temp_file:
+                editor_script = temp_file.name
+                temp_file.write(
+                    f"""#!/bin/bash
+                    todo_file=".git/rebase-merge/git-rebase-todo"
+                    tmp_file="$todo_file.tmp"
+
+                    # Copy todo_file to a temporary file
+                    cp "$todo_file" "$tmp_file"
+                    printf "%s\\n" "/^pick {commit.hexsha[:9]}/ m1" "wq" | ed -s "$tmp_file" 
+                    printf "%s\\n" "/^pick {commit.hexsha[:9]} /s//squash {commit.hexsha[:9]} /" "wq" | ed -s "$tmp_file"
+                    mv "$tmp_file" "$todo_file"
+                    """
+                )
+            os.chmod(editor_script, 0o755)
+            result = subprocess.run(
+                f"GIT_SEQUENCE_EDITOR='{editor_script}' GIT_EDITOR=true git rebase -i {base_commit}",
+                capture_output=True,
+                shell=True,
+            )
+        else:
+            command = f"GIT_SEQUENCE_EDITOR='sed -i \"s/^pick {commit.hexsha[:9]} /squash {commit.hexsha[:9]} /\"' GIT_EDITOR=true git rebase -i {base_commit}"
+            result = subprocess.run(command, capture_output=True, shell=True)
+        output = result.stdout.decode("utf-8")
+        if editor_script:
+            os.remove(editor_script)
+
+        if "CONFLICT" in output:
+            self._print(f"\n{bc.FAIL}ERROR: A conflict occurs{bc.ENDC}")
+            self._print(
+                "\n ⚠️You can't squash those commits together and they should be left as is"
+            )
+            self._abort_rebase()
+            return False
+        click.echo(
+            click.style(
+                "✨ Done! Successfully squashed.",
+                fg="green",
+                bold=True,
+            )
+        )
+        return True
+
+    def _abort_rebase(self):
+        self.app.repo.git.rebase("--abort")
+
+    def is_skipped_commit(self, commit):
+        return any(
+            skipped_commit.__eq__(commit) for skipped_commit in self.skipped_commits
```

Comment:
```suggestion
        return commit in self.skipped_commits
```

Reply:
python will properly iterate over all objects in `self.skipped_commits` and call `.__eq__(commit)` on them